# HYPOTHESIS TESTING

In [38]:
import pandas as pd
import numpy as np
import scipy

# QUES1

## Business Problem: Detecting difference in the diameter of the cutlet of two units or not.

In [39]:
data=pd.read_csv("Cutlets.csv")
df=pd.DataFrame(data)
df.head()

Unit A  Unit B
0  6.8090  6.7703
1  6.4376  7.5093
2  6.9157  6.7300
3  7.3012  6.7878
4  7.4488  7.1522

### alpha=0.05 and CI=0.95

### To check Unit A and Unit B is normal or not go for shapiro test

In [40]:
from scipy import stats
stats.shapiro(df["Unit A"])

ShapiroResult(statistic=0.9649458527565002, pvalue=0.3199819028377533)

### therefore for unit A p-value is greater than alpha value so, Unit A is normal.

In [41]:
stats.shapiro(df["Unit B"])

ShapiroResult(statistic=0.9727300405502319, pvalue=0.5224985480308533)

### therefore for unit B p-value is greater than alpha value so, Unit B is normal.

### Here, Ho [null hypo]= There is no difference in the diameter of cutlet, whereas
### H1 [alt hypo]= There is a difference in the diameter of cutlet.

In [42]:
stats.ttest_ind(df["Unit A"],df["Unit B"])

Ttest_indResult(statistic=0.7228688704678061, pvalue=0.47223947245995)

## p-value for the data is 0.48 which is less than 0.05 so Ho is accepted i.e., There is no difference in the diameter of cutlet.

# QUES2

## Business Problem: To determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories or not.

In [43]:
data=pd.read_csv("LabTAT.csv")
df=pd.DataFrame(data)
df.head()

Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0        185.35        165.53        176.70        166.13
1        170.49        185.91        198.45        160.79
2        192.77        194.92        201.23        185.18
3        177.33        183.00        199.61        176.42
4        193.41        169.57        204.63        152.60

### alpha=0.05 and CI=0.95

### To check Unit A and Unit B is normal or not go for shapiro test

In [44]:
stats.shapiro(df["Laboratory 1"])

ShapiroResult(statistic=0.9901824593544006, pvalue=0.5506953597068787)

In [45]:
stats.shapiro(df["Laboratory 2"])

ShapiroResult(statistic=0.9936322569847107, pvalue=0.8637524843215942)

In [46]:
stats.shapiro(df["Laboratory 3"])

ShapiroResult(statistic=0.9886345267295837, pvalue=0.4205053448677063)

In [47]:
stats.shapiro(df["Laboratory 4"])

ShapiroResult(statistic=0.9913753271102905, pvalue=0.6618951559066772)

### therefore p-value for above data is greater than alpha so it is normal

### Here, Ho [null hypo]= There is no difference in the average Turn Around Time (TAT) of reports of the laboratories , whereas
### H1 [alt hypo]= There is a difference in the average Turn Around Time (TAT) of reports of the laboratories.

### To check average inside my groups is equal or not we need to go for ANOVA test. 

In [48]:
import scipy.stats as stats
stats.f_oneway(df.iloc[:,0], df.iloc[:,1],df.iloc[:,2],df.iloc[:,3])

F_onewayResult(statistic=118.70421654401437, pvalue=2.1156708949992414e-57)

## Here, p-value is greater than 0.05 so H1 is accepted i.e. There is a difference in the average Turn Around Time (TAT) of reports of the laboratories.

# QUES3

## Business Problem: All proportion among male and female are equal or not.

In [49]:
data=pd.read_csv("BuyerRatio.csv")
df=pd.DataFrame(data)
df.head()

Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750

### Here, Ho [null hypo]= All proportions are equal, whereas
### H1 [alt hypo]= All proportions are not equal.

### When there are more than two groups we go for Chi-square testing.

In [50]:
stats.chi2_contingency([df["East"],df["West"],df["North"],df["South"]])

(1.5959455386610577,
 0.6603094907091882,
 3,
 array([[  42.76531299,  442.23468701],
        [ 146.81287862, 1518.18712138],
        [ 131.11756787, 1355.88243213],
        [  72.30424052,  747.69575948]]))

### 1.59 = test stats
## 0.66 = p-value
### array([[  42.76531299,  442.23468701], #Expected Frequency
###        [ 146.81287862, 1518.18712138],
###        [ 131.11756787, 1355.88243213],
###        [  72.30424052,  747.69575948]]))

# Here p-value=0.66 which is greater than 0.05 so Ho is accepted i.e., All proportions are equal.

# QUES4

## Business Problem: The defective % varies among 4 centers or not.

In [51]:
data=pd.read_csv("Customer order form.csv")
df=pd.DataFrame(data)
df.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

### Here, Ho [null hypo]= Defective % does not varies by center, whereas
### H1 [alt hypo]= Defective % varies by center.

### When there are more than two groups we go for Chi-square testing.

In [52]:
stats.chi2_contingency([df["Phillippines"].value_counts(),df["Indonesia"].value_counts(),df["Malta"].value_counts(),df["India"].value_counts()])

(3.8589606858203545,
 0.2771020991233144,
 3,
 array([[271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25],
        [271.75,  28.25]]))

### 3.85 = test stats
## 0.277 = p-value
### array([[271.75,  28.25], #Expected Frequency
###        [271.75,  28.25],
###        [271.75,  28.25],
###        [271.75,  28.25]]))

# Here p-value=0.277 which is greater than 0.05 so Ho is accepted i.e.,Defective % does not varies by center .